app/src/controllers/object.js

```diff
@@ -91,6 +90,47 @@ const controller = {
     }
   },
 
+  /**
+   * @function addTags
+   * Adds the tag set to the requested object
+   * @param {object} req Express request object
+   * @param {object} res Express response object
+   * @param {function} next The next callback function
+   * @returns {function} Express middleware function
+  */
+  async addTags(req, res, next) {
+    try {
+      const objId = addDashesToUuid(req.params.objId);
+      const objPath = getPath(objId);
+      const { versionId, ...newTags } = req.query;
+      const objectTagging = await storageService.getObjectTagging({ filePath: objPath, versionId });
+
+      // Join new and existing tags then filter duplicates
+      let newSet = Object.entries(newTags).map(([k, v]) => ({ Key: k, Value: v }));
+      if (objectTagging.TagSet) newSet = newSet.concat(objectTagging.TagSet);
+      newSet = newSet.filter((v, i, a) => a.findIndex(v2 => (v2.Key === v.Key)) === i);
```

Comment:
It may be worthwhile having more descriptive variable names in the anonymous function as they aren't innately clear what they represent without re-reading multiple times. Key/value only is fine making them k and v as that's pretty clear, but here not too sure what "a" is.